### PR TITLE
Add support for job token

### PR DESCRIPTION
**Pull request by sathieu**
See https://docs.gitlab.com/ee/api/jobs.html#get-job-artifacts for usage

### CODE REVIEW

File: docs/cli.rst [hunk 1]
```diff
@@ -108,7 +110,8 @@ limited permissions.
    * - ``http_password``
      - Password for optional HTTP authentication
 
-__ https://docs.gitlab.com/ce/user/profile/personal_access_tokens.html
+.. _pat: https://docs.gitlab.com/ce/user/profile/personal_access_tokens.html
+.. _job-token: https://docs.gitlab.com/ee/api/jobs.html#get-job-artifacts
```
Comment (max-wittig) [inline docs/cli.rst]: let's keep links with `CE` here to be consistent

**Comment by sathieu** [comment]
@max-wittig All done. Please merge.

File: gitlab/__init__.py [hunk 2]
```diff
@@ -346,9 +350,17 @@ def _construct_url(self, id_, obj, parameters, action=None):
...
     def _set_auth_info(self):
-        if self.private_token and self.oauth_token:
+        if (
+            self.private_token
+            and self.oauth_token
+            or self.private_token
+            and self.job_token
+            or self.oauth_token
+            and self.job_token
+        ):
```
Comment (max-wittig) [inline gitlab/__init__.py]: This seems confusing. 

What about a true XOR?

```python
if sum(bool(arg) for arg in [self.private_token, self.oauth_token, self.job_token]) == 1
```

Comment (sathieu) [reply gitlab/__init__.py]: Done (it's `!= 1`).

Comment (max-wittig) [reply gitlab/__init__.py]: ah, you're right!

**Comment by sathieu** [comment]
Thanks

**Pull request merged**